pkg/operation/botanist/component/istio/networkpolicies.go

```diff
@@ -0,0 +1,511 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
+
+import (
+	"fmt"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/coredns"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/kubeapiserver"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/nodelocaldns"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnauthzserver"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnseedserver"
+
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	istiodAppLabelValue = "istiod"
+)
+
+// IstioNetworkPolicyValues contains deployment parameters for the istio-ingress and istio-system network policies.
+type IstioNetworkPolicyValues struct {
+	// NodeLocalIPVSAddress is the CIDR of the node-local IPVS address.
+	NodeLocalIPVSAddress *string
+	// DNSServerAddress is the CIDR of the usual DNS server address.
+	DNSServerAddress *string
+}
+
+type networkPolicyTransformer struct {
+	name      string
+	transform func(*networkingv1.NetworkPolicy) func() error
+}
+
+func getIstioSystemNetworkPolicyTransformers(values IstioNetworkPolicyValues) []networkPolicyTransformer {
+	return []networkPolicyTransformer{
```

Comment:
How is this working? There is a deny all policy for egress and ingress, but only ingress policies. I would have expected that istiod needs to reach out to the seed`s kube-apiserver, i.e. there would be a egress policy required.

Reply:
Yes, that was a bit tricky, as it's not trivial to determine the endpoints for the api server. However, as there is already such a networkpolicy for garden namespace and shoot namespaces, I just added istio-system there:https://github.com/gardener/gardener/blob/4377d31f71570bd88b05752c02b04436ba9621d3/pkg/gardenlet/controller/networkpolicy/namespace_control.go#L82
This needs an added label to the istiod pods. 

But I didn't want to add a dns resolve here.